src/Features/Core/Portable/DocumentationComments/AbstractDocumentationCommentFormattingService.cs

```diff
@@ -527,6 +621,23 @@ private static void AppendTextFromAttribute(FormatterState state, XElement eleme
                 new SymbolDisplayPart(kind, symbol: null, text: TrimCrefPrefix(crefValue)));
         }
 
+        internal static IEnumerable<SymbolDisplayPart> TypeParameterRefToSymbolDisplayParts(
+            string crefValue, ISymbol typeResolutionSymbol, int position, SemanticModel semanticModel, SymbolDisplayFormat format = null)
+        {
+            if (semanticModel != null)
+            {
+                var typeParameterIndex = typeResolutionSymbol.OriginalDefinition.GetAllTypeParameters().IndexOf(tp => tp.Name == crefValue);
+                if (typeParameterIndex > -1)
+                {
+                    return typeResolutionSymbol.GetAllTypeArguments()[typeParameterIndex].ToMinimalDisplayParts(semanticModel, position, format);
```

Comment:
need to make sure that typeParmeterIndex is within the range of type arguments returned.  

also, we need lots of tests for `<typeparamref ...` with cases of nested generics.

Reply:
do you mean unit tests by saying "lots of tests"? i haven't found any for these files.

Reply:
yes, i'm saying unit tests.  @sharwell do you know where tests should go for this PR?

Reply:
added a check on GetAllTypeArguments().Length